### PR TITLE
feat: proper option parser for replace modifier

### PR DESCRIPTION
**Pull request by seia-soto**
This new option parser gives you an ability to parse the option value depending on the option key.

### CODE REVIEW

File: packages/adblocker/src/filters/network.ts
Comment (remusao) [inline packages/adblocker/src/filters/network.ts:1]: Could you add more unit tests to extend the coverage (in particular to cases which were not covered before)?

**Comment by remusao** [comment]
Could you please add some comments/documentation about how the new parsing logic works (as well as examples of cases which were not working before and are working now, ideally in the form of unit tests)?

File: packages/adblocker/src/filters/network.ts
Comment (seia-soto) [reply packages/adblocker/src/filters/network.ts:1]: This PR addresses the filter parsing failures happening on the parent PR: https://github.com/ghostery/adblocker/pull/3897 . Some filters including an comma sign inside of their option value interfere the current parsing logic.

**Comment by seia-soto** [comment]
I added some comments to the function. For the unit tests, I don't think we'll need an additional unit test as this function can cover all of the existing filter tests we currently do. Also, this changes are required to fix the failure of replace modifier option value parsing which is necessary for the upstream PR.

The only change from the last update is adding comments on the functions. Please, let me know if you have some points that seem like not reasonable. I think this is a real hot-path in the library and should be really fast. For the benchmarking, I'll try to do in a separate branch since the upstream branch had to be rebased to leverage the benchmarking fix which was merged in master.

**Pull request merged**